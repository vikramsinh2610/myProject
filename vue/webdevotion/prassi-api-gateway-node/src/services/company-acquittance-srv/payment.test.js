const Payment = require('./payment');

describe('Payment', () => {
  test('should have these properties', () => {
    const payment = new Payment({
      _id: 'ID',
      type: 'subscription',
      contractId: 'ContractId',
      premiumNet: 1234,
      premiumGross: 2345,
      installment: 2,
      payin: 5500,
      installmentDate: new Date(),
      productivePeriodYear: 2018,
      productivePeriodMonth: 8,
    });
    expect(payment).toHaveProperty('_id');
    expect(payment).toHaveProperty('contractId');
    expect(payment).toHaveProperty('type');
    expect(payment).toHaveProperty('premiumNet');
    expect(payment).toHaveProperty('premiumGross');
    expect(payment).toHaveProperty('installment');
    expect(payment).toHaveProperty('payin');
    expect(payment).toHaveProperty('installmentDate');
    expect(payment).toHaveProperty('productivePeriodYear');
    expect(payment).toHaveProperty('productivePeriodMonth');
    expect(payment).toHaveProperty('ok');
    expect(payment).toHaveProperty('calculatedPayin');
  });

  test('should have JSON Schema', () => {
    expect(Payment.getJSONSchema()).toMatchSnapshot();
  });
});
