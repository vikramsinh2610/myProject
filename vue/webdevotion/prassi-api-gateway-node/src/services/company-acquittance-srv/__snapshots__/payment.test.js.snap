// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should have JSON Schema 1`] = `
Object {
  "properties": Object {
    "_id": Object {
      "description": "ID of payment",
      "type": "string",
    },
    "alreadyConfirmed": Object {
      "description": "Is payment ok?",
      "type": "boolean",
    },
    "alreadyPaid": Object {
      "description": "Is payment ok?",
      "type": "boolean",
    },
    "calculatedPayin": Object {
      "description": "Calculated payin of installment",
      "type": "integer",
    },
    "contractId": Object {
      "description": "ID of contract",
      "type": "string",
    },
    "errorPayin": Object {
      "description": "Is payment ok?",
      "type": "boolean",
    },
    "incomePracticeId": Object {
      "description": "Linked income practice id",
      "type": "string",
    },
    "installment": Object {
      "description": "Installment number",
      "type": "integer",
    },
    "installmentDate": Object {
      "description": "Date of payment",
      "type": "string",
    },
    "manuallyModified": Object {
      "description": "Is payment ok?",
      "type": "boolean",
    },
    "notFoundBase": Object {
      "description": "Is payment ok?",
      "type": "boolean",
    },
    "notFoundPractice": Object {
      "description": "Is payment ok?",
      "type": "boolean",
    },
    "ok": Object {
      "description": "Is payment ok?",
      "type": "boolean",
    },
    "oldCalculatedPayin": Object {
      "description": "Calculated payin of installment",
      "type": "integer",
    },
    "oldPayin": Object {
      "description": "Payment payin",
      "type": "integer",
    },
    "payin": Object {
      "description": "Payment payin",
      "type": "integer",
    },
    "practiceId": Object {
      "description": "ID of practice",
      "type": "string",
    },
    "premiumGross": Object {
      "description": "Practice gross premium",
      "type": "integer",
    },
    "premiumNet": Object {
      "description": "Practice net premium",
      "type": "integer",
    },
    "productivePeriodMonth": Object {
      "description": "Productive period month of installment",
      "type": "integer",
    },
    "productivePeriodYear": Object {
      "description": "Productive period year of installment",
      "type": "integer",
    },
    "select": Object {
      "type": "string",
    },
    "status": Object {
      "description": "Status description",
      "type": "string",
    },
    "type": Object {
      "description": "Type of payment",
      "enum": Array [
        "subscription",
        "income",
        "additional-income",
      ],
      "type": "string",
    },
  },
  "type": "object",
}
`;
